File: src/main/java/com/sfh/pokeRogueBot/Application.java
package com.sfh.pokeRogueBot;

import com.sfh.pokeRogueBot.bot.Bot;
import com.sfh.pokeRogueBot.bot.SimpleBot;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;

@SpringBootApplication
@Slf4j
public class Application {

	public static void main(String[] args) {
		ConfigurableApplicationContext context = SpringApplication.run(Application.class, args);
		try{
			Bot bot = context.getBean(SimpleBot.class);
            bot.start();
        } catch (Exception e) {
			log.error("Error starting the bot: " + e.getMessage(), e);
		}
	}
}

File: src/main/kotlin/com/sfh/pokeRogueBot/Application.kt
package com.sfh.pokeRogueBot

import com.sfh.pokeRogueBot.config.GameSettingsConfig
import com.sfh.pokeRogueBot.config.WaitConfig
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication

@EnableConfigurationProperties(WaitConfig::class, GameSettingsConfig::class)
@SpringBootApplication
class Application

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}
